Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector.py

```diff
@@ -169,28 +241,63 @@ def create_api_call(host: str, integration_key: str, secrete_key: str):  # pragm
     return client
 
 
-def main():  # pragma: no cover
+def parse_events(authentication_evetns: list):
+    """
+    Adds the parsing rule of the _time to each of duo events
+    """
+    for event in authentication_evetns:
+        event["_time"] = event.get("isotimestamp")
+
+    return authentication_evetns
+
+
+def parse_mintime(last_run: float) -> tuple:
+    """Returns the last run percision of 10 digits for v1 and 13 digits for v2"""
+    last_run_v1 = int(last_run)
+    last_run_v2 = int(last_run * 1000)
+    return last_run_v1, last_run_v2
+
+
+def validate_request_order_array(logs_type_array: list) -> Any:
+    """Validates that all the inputs of the log_type_array are valid."""
+    for value in logs_type_array:
+        if value not in [LogType.ADMINISTRATION, LogType.AUTHENTICATION, LogType.TELEPHONY]:
+            return value
+    return True
```

Comment:
```suggestion
    """Validates that all the inputs of the log_type_array are valid."""
    wrong_values = []
    for value in logs_type_array:
        if value not in [LogType.ADMINISTRATION, LogType.AUTHENTICATION, LogType.TELEPHONY]:
            wrong_values.append(value)
   if not wrong_values:
       return True
  else:
      return ','.join(wrong_values)
```
You probably want to return all the wrong values